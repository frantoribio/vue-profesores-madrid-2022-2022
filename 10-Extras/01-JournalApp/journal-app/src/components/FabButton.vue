<template>
  <!-- Me voy a inventar un evento solo para testear -->
  <button
    class="btn btn-primary btn-circle shadow-lg"
    @click="$emit('onClick')"
  >
    <Icon
      :icon="props.icon"
      class="h-6 w-6"
    />
  </button>
</template>

<script setup>
  import { Icon } from '@iconify/vue'

  const props = defineProps({
    icon: {
      type: String,
      required: true,
    },
  })

  defineEmits(['onClick'])

  // console.log('props', props.icon)
</script>

<style lang="scss" scoped></style>
