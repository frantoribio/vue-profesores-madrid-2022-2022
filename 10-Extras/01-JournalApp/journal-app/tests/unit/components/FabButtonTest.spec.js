import { beforeEach, describe, expect, test } from 'vitest'

import FabButton from '@/components/FabButton.vue'

import { shallowMount } from '@vue/test-utils'

// Suite de Test
describe('Component -> FabButton', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(FabButton, {
      props: {
        icon: 'testing-icon',
      },
    })
  })

  test('Debe hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  // Testeamos propiedades
  test('Debe mostrar las propiedades del boton', () => {
    expect(wrapper.html()).toContain('testing-icon')
  })

  // Testeamos el evento
  test('Debe emitir ele vento onClick al hacer click', () => {
    const button = wrapper.find('button')
    // pulsamos el evento click
    button.trigger('click')
    // vemos si emitimos el evento onClick
    expect(wrapper.emitted('onClick')).toHaveLength(1)
  })
})
