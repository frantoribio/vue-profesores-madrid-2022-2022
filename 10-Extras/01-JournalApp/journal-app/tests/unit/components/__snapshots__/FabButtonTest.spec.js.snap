// Vitest Snapshot v1

exports[`Component -> FabButton > Debe hacer match con el snapshot 1`] = `
"<!-- Me voy a inventar un evento solo para testear -->
<button class=\\"btn btn-primary btn-circle shadow-lg\\">
  <icon-stub icon=\\"testing-icon\\" class=\\"h-6 w-6\\"></icon-stub>
</button>"
`;
